### ISSUE TITLE
ignoreUnknownKeys doesn't work in decodeFromJsonElement

### ISSUE THREAD

**Issue by bkadatz** [Kotlin/kotlinx.serialization#1694]
**Describe the bug**

`ignoreUnknownKeys` works only for `decodeFromString` but doesn't work for `decodeFromJsonElement`.

**To Reproduce**

```
@Serializable
data class Foo(val a: Int)
```
Then try to parse this in various ways:
```
val format = Json { ignoreUnknownKeys = true }
val json = "{ \"a\": 1, \"b\": 2 }"
val fooSuccess = format.decodeFromString<Foo>(json) // works

val element = format.parseToJsonElement(json)
val fooException = format.decodeFromJsonElement<Foo>(element) // throws exception
```
This isn't a big deal when it comes to simple JSON, but when dealing with APIs you often want to use only part of the response. For example, an API might return:
```
{
   "data": {
      // object you really care about
   }
}
```
and it's handy to extract the object via:
```
val element = format.parseToJsonElement(json).jsonObject["data"]
```
and then decode the element to your data class. There is a workaround, however:
```
val elementJson = format.parseToJsonElement(json).jsonObject["data"].toString()
val foo = format.decodeFromString<Foo>(elementJson)
```
but this seems like an unnecessary step.

**Expected behavior**

Since both methods are doing a "decode", I expect `ignoreUnknownKeys` to also work for `decodeFromJsonElement`, not just `decodeFromString`.

**Environment**
 - Kotlin version: 1.4.30
 - Library version: 1.4.0
 - Kotlin platforms: JVM
 - Gradle version: 7.1.1

**Comment by qwwdfsad** [comment]
Could you please ensure that it is not a problem in your setup?

We do not have the 1.4.0 version and the following code works as expected:
```
class Repro {
    @Serializable
    data class Foo(val a: Int)


    @Test
    fun doTest() {
        val format = Json { ignoreUnknownKeys = true }
        val json = "{ \"a\": 1, \"b\": 2 }"
        val fooSuccess = format.decodeFromString<Foo>(json) // works
        println(fooSuccess)
        val element = format.parseToJsonElement(json)
        val fooException = format.decodeFromJsonElement<Foo>(element) // throws exception
        println(fooException)
    }
}

```

**Comment by bkadatz** [comment]
You're right. I was still using the ancient `1.0.0` library version as the dependency; the 1.4.0 was actually the Kotlin version. I updated this to use `1.3.0-RC` and 1.5.31 as shown [here](https://kotlinlang.org/docs/serialization.html#example-json-serialization) and it's working as expected now.

It turns out I was using the outdated instructions at https://blog.jetbrains.com/kotlin/2020/10/kotlinx-serialization-1-0-released/ under "Adding kotlinx.serialization to your project" and didn't check the current docs. Mea culpa. Closing this as it's not an issue in the current version.
